File: 백준/Silver/10773. 제로/제로.cc
#include <bits/stdc++.h>

using namespace std;

int K, num, ans;
stack<int> s;

int main() {
    cin.tie(0) -> ios::sync_with_stdio(0);
    
    cin >> K;
    
    for(int i = 0; i < K; i++) {
        cin >> num;
        if(num != 0) {
            s.push(num);
        } else {
            s.pop();
        }
    }
    
    while(!s.empty()) {
        ans += s.top();
        s.pop();
    }
    
    cout << ans;
    
    return 0;
}

File: 백준/Silver/10773. 제로/제로.py
K = int(input())
arr = []

for _ in range(K):
    num = int(input())
    if num == 0:
        arr.pop()
    else:
        arr.append(num)

if len(arr) == 0:
    print(0)
else:
    print(sum(arr))